api/src/services/rapid-pro.js

```diff
@@ -0,0 +1,240 @@
+const _ = require('lodash');
+const request = require('request-promise-native');
+const secureSettings = require('@medic/settings');
+const logger = require('../logger');
+const config = require('../config');
+const db = require('../db');
+
+/**
+ * Mapping of RapidPro message/broadcast statuses to CHT-Core message states
+ * Keys represent RapidPro statuses, `state` fields represent CHT-Core states
+ * https://github.com/rapidpro/rapidpro/blob/28d3215d57c152af0a71798a4ffe9351d10a3e95/temba/msgs/models.py#L57
+ */
+const STATUS_MAP = {
+  // success
+  queued: { success: true, state: 'received-by-gateway', detail: 'Queued' },
+  wired: { success: true, state: 'forwarded-by-gateway', detail: 'Wired' },
+  sent: { success: true, state: 'sent', detail: 'Sent' }, // todo Reviewer: should this be a final state?????
+  delivered: { success: true, state: 'delivered', detail: 'Delivered', final: true },
+  resent: { success: true, state: 'sent', detail: 'Resent' },
+
+  // failure
+  errored: { success: false, state: 'received-by-gateway', detail: 'Errored' }, // sending will be retried
+  failed: { success: false, state: 'failed', detail: 'Failed', final: true },
+};
+const BATCH_SIZE = 25;
+
+const nonFinalStates = Object
+  .values(STATUS_MAP)
+  .filter(status => !status.final)
+  .map(status => status.state);
+
+const getCredentials = () => {
+  // Supports self-hosted RapidPro instances, but defaults to textit.in
+  const smsSettings = config.get('sms');
+  const configuredUrl = smsSettings &&
+                        smsSettings.rapidPro &&
+                        smsSettings.rapidPro.url;
+  const textitUrl = 'https://textit.in';
```

Comment:
I'm not sure about adding a default value in here. Frankly for PII reasons we'd rather everyone used our RapidPro server than TextIt. I would require it to be configured manually.

Reply:
Ok cool, I'll remove!